config/dev.exs

```diff
@@ -19,7 +19,7 @@ config :skate, Skate.Detours.TripModificationPublisher, start: false
 config :skate, SkateWeb.Endpoint,
   https: [
     ip: {0, 0, 0, 0},
-    port: 4000,
+    port: System.get_env("PORT") || 4000,
```

Comment:
> But why not commit those changes to make my dev life easier? Hence my curiosity about what configurations do / do not get committed to the dev config file

I don't _personally_ think we should be using http for local dev, because all of Skate happens in https in prod, so I ignore the message from Phoenix because I think it's wrong. Additionally, we now use features in Skate that you _cannot_ use in a insecure environment (meaning served over http), this was first introduced when we started using the location information from the browser because browsers don't let you access that without https.

Having said this, if you agree, I think I should add a comment noting this :)

> 👀 what is the purpose of this change to only dev?

- For `test` envs, we don't need to think about this (afaik).
- For `prod`, this seems to already be configured to [use the port variable if it's present](https://github.com/mbta/skate/blob/03652e9cba76a91a8a224d0ffff440d984e4cb97/config/prod.exs#L26), but, this happens in `prod.exs` not `runtime.exs` so this variable is only respected at build time, not when it's deployed.
- For `dev`, when running multiple instances of Skate, it's necessary to change the port because two instances cannot use the same port, which is why I added this now.
 
I'd be open to considering moving this change into `runtime.exs` but I think we'd need to consider it a bit more than this change because that could have production implications, unless we gated it on `Mix.env() == :dev`.